Graded/Modality/Instances/Affine/Bad.agda

```diff
@@ -15,28 +14,38 @@ module Graded.Modality.Instances.Affine.Bad
   -- The modality variant.
   (variant : Modality-variant)
   (open Graded.Modality.Instances.Affine variant)
-  (open Graded.Modality.Dedicated-nr bad-affine-modality)
-  (TR : Type-restrictions bad-affine-modality)
+  (TR : Type-restrictions affineModality)
   (open Type-restrictions TR)
-  (UR : Usage-restrictions bad-affine-modality)
+  (UR : Usage-restrictions affineModality)
   -- It is assumed that "Π 𝟙 , 𝟘" is allowed.
   (Π-𝟙-𝟘 : Π-allowed 𝟙 𝟘)
-  -- There is a dedicated nr function.
-  ⦃ has-nr : Dedicated-nr ⦄
   where
 
+open import Graded.Restrictions affineModality
+open import Graded.Modality Affine
+open import Graded.Usage.Restrictions.Natrec affineModality
+
+private
+  module M = Modality affineModality
+
+  -- The "bad" nr function is used
+  UR′ = nr-available-UR zero-one-many-greatest-star-nr UR
+  open Usage-restrictions UR′
+  instance
+    has-nr : Has-nr M.semiring-with-meet
```

Comment:
Is `has-nr` only used in the implementation of `nr-avail`? In that case it might make sense to inline `has-nr` (and rename `nr-avail`). However, I tried to do this, and for some reason Agda then failed to instantiate some instance arguments below, even though those arguments could be instantiated using `nr-avail`. I think this should be investigated.

Similar comments apply to other modules below as well.